src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_x509.c

```diff
@@ -260,7 +260,7 @@ void* AndroidCryptoNative_X509PublicKey(jobject /*X509Certificate*/ cert, PAL_Ke
             keyHandle = AndroidCryptoNative_NewEcKeyFromPublicKey(env, key);
             break;
         case PAL_DSA:
-            keyHandle = NULL;
+            keyHandle = (*env)->NewObject(env, g_keyPairClass, g_keyPairCtor, key, NULL);
```

Comment:
I won't say that this is incorrect... but I'll say that it stands out as notably different from the above and below case blocks.

Reply:
Yes, it does stand out. I'll add an abstraction layer.